CHANGES.txt

```diff
@@ -11,6 +11,7 @@ Fixed: GITHUB-2637: Upgrade to JDK11 as the minimum JDK requirements (Krishnan M
 Fixed: GITHUB-2734: Keep the initial order of listeners (Andrei Solntsev)
 Fixed: GITHUB-2359: Testng @BeforeGroups is running in parallel with testcases in the group (Anton Velma)
 Fixed: Possible StringIndexOutOfBoundsException in XmlReporter (Anton Velma)
+Fixed: AfterGroups methods could be invoked before all tests for defined groups in case of multiple groups for AfterGroups method (Anton Velma)
```

Comment:
For the sake of tracking, request you to please help file a bug and link it here so that we can have way to trace back the change, if required in the future.

Reply:
@velma - Can you please update the `CHANGES.txt` to include a reference to the github issue https://github.com/cbeust/testng/issues/2754